tests/contrib/pydantic/test_pydantic.py

```diff
@@ -324,3 +330,36 @@ async def test_validation_error(client: Client):
                 task_queue=task_queue_name,
                 result_type=tuple[int],
             )
+
+
+class RestrictedProxyFieldsModel(BaseModel):
+    datetime_field: datetime
+    path_field: pathlib.Path
+
+
+def test_model_instantiation_from_restricted_proxy_values():
+    restricted_path_cls = _RestrictedProxy(
+        "Path",
+        pathlib.Path,
+        RestrictionContext(),
+        SandboxMatcher(),
+    )
+    restricted_datetime_cls = _RestrictedProxy(
+        "datetime",
+        datetime,
+        RestrictionContext(),
+        SandboxMatcher(),
+    )
+
+    restricted_path = restricted_path_cls("test/path")
+    restricted_datetime = restricted_datetime_cls(2025, 1, 2, 3, 4, 5)
+
+    assert type(restricted_path) is _RestrictedProxy
+    assert type(restricted_datetime) is _RestrictedProxy
+
+    p = RestrictedProxyFieldsModel(
+        datetime_field=restricted_datetime,  # type: ignore
+        path_field=restricted_path,  # type: ignore
```

Comment:
The restricted `pathlib.Path` instance is accepted by pydantic but the restricted `datetime.datetime` is not:

```
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for RestrictedProxyFieldsModel
E       datetime_field
E         Input should be a valid datetime [type=datetime_type, input_value=datetime.datetime(2025, 1, 2, 3, 4, 5), input_type=_RestrictedProxy]
E           For further information visit https://errors.pydantic.dev/2.10/v/datetime_type
```

Reply:
This seems to be due to the fact that `datetime` has a custom validation method in Pydantic's Rust layer: https://github.com/pydantic/pydantic-core/blob/741961c05847d9e9ee517cd783e24c2b58e5596b/src/input/input_python.rs#L548-L582
and `_RestrictedProxy` does not pass the `self.downcast::<PyDateTime>` test which leads to the validation failure.

UUID does also: https://github.com/pydantic/pydantic-core/blob/741961c05847d9e9ee517cd783e24c2b58e5596b/src/validators/uuid.rs#L95-L154 but `_RestrictedProxy` does not cause it to fail. And `pathlib.Path` does not seem to have a dedicated validator in `pydantic_core`.